draft-ietf-tls-esni.md

```diff
@@ -787,6 +815,48 @@ where HKDF-Expand-Label and HKDF-Extract are as defined in {{RFC8446}}. The
 value of ServerHello.random[0:24] is generated as usual by invoking a secure
 random number generator (see {{RFC8446}}, Section 4.1.2).
 
+When confirming ECH acceptance, the backend server SHOULD reply to the
+"handshake_padding" extension solicited by the client, computing the padding
+length as described in {{server-padding}}.
+
+### Recommended Padding Scheme {#server-padding}
+
+It is RECOMMENDED that the backend server compute the padding length for the
+"handshake_padding" extension in its EncryptedExtensions message
+({{handshake-padding}}) based on the following values:
+
+1. Its Certificate message.
+1. The "application_layer_protocol_negotiation" extension offered by the client
+   in the ClientHelloInner.
+
+Let certificate_padding_length = max_certificate_length - certificate_length,
+where:
+
+* max_certificate_length is equal to the length of the largest Certificate
+  message that would be sent by any backend server;
+* and certificate_length is the length of the Certificate message selected by
```

Comment:
Is this the Certificate message after compression, or before? I assume the former, though it would be good to clarify.

Reply:
The former. More precisely, what's intended is the Certificate message as it appears on the wire. I'll clarify this. Can you provide me a reference for certificate compression? I'm unfamiliar with that spec.

Reply:
Sure! https://datatracker.ietf.org/doc/draft-ietf-tls-certificate-compression/